model/atmosphere/dycore/README.md

```diff
@@ -4,6 +4,66 @@
 
 Contains code ported from ICON `src/atm_dyn_iconam`, which is the dynamical core of the ICON model.
 
+## Attention
+
+Variables in ICON can either sit on full levels or half levels. Let's denote the number of full levels is nlev.
+Full-level variables have nlev levels in vertical dimension, while half-level variables have nlev+1 in vertical dimension.
+Do not output full-level and half-level variables together when running a gt4py program if you need to access the nlev+1 level because it will cause random behavior (most probably illegal memory access).
```

Comment:
```suggestion
Do not output full-level and half-level variables together when running a GT4Py program over nlev+1 levels because the nlev+1 level will also be written for a full-level variable, which is out-of-bounds.
```
The behavior is not random, only how it is observable. ;)

Reply:
Okay. Thanks.